### PR TITLE
provider/ns1: Ensure provider checks for credentials

### PR DESCRIPTION
**Pull request by pashap**
Fixes: https://github.com/hashicorp/terraform/issues/12833

I somehow never added a check for the ns1 api key, this PR mimics other providers `Config` structures. Also, this PR ensures defaults for the ns1 api endpoint and whether to ignore ssl in the client(use terraform often for local development).

### CODE REVIEW

File: builtin/providers/ns1/provider.go [hunk 1]
```diff
@@ -49,22 +44,13 @@ func Provider() terraform.ResourceProvider {
...
-		decos = append(decos, ns1.SetEndpoint(v.(string)))
-	}
-	if _, ok := d.GetOk("ignore_ssl"); ok {
-		tr := &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-		}
-		httpClient.Transport = tr
+	config := Config{
+		Key:       d.Get("apikey").(string),
+		Endpoint:  d.Get("endpoint").(string),
```
Comment (stack72) [inline builtin/providers/ns1/provider.go]: I think we  should follow the pattern of 

```
if v, ok := d.GetOk("endpoint"); ok {
  config.Endpoint = v.(string)
}
```

when we are adding optional parameters to the config block

Sound ok?

Comment (pashap) [inline builtin/providers/ns1/provider.go]: Great point, i wasnt thinking when moving away from `GetOk`. Making fix now.

**Comment by pashap** [comment]
I was wrong about what caused the error, it was a [bug](https://github.com/ns1/ns1-go/pull/34) in the ns1-go client that wasnt properly handling 401s. Updating the ns1-go client via govendor and running acctests then will include in this PR

**Comment by stack72** [comment]
Thanks for the work here @pashap :)

**Comment by kian** [comment]
Seconded, thanks for this work @pashap!

**Comment by kian** [comment]
Also noticed the NS1 provider seems to think resources are new/modified between sequential `terraform apply` calls.  @pashap / @stack72 would you like me to fill out a new issue for that or leave the notes here?

**Comment by stack72** [comment]
LGTM! The build failure is not related to this - it is a timing out test in the core

```
% make testacc TEST=./builtin/providers/ns1
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/03/23 11:54:41 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/ns1 -v  -timeout 120m
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccDataFeed_basic
--- PASS: TestAccDataFeed_basic (14.44s)
=== RUN   TestAccDataFeed_updated
--- PASS: TestAccDataFeed_updated (17.92s)
=== RUN   TestAccDataSource_basic
--- PASS: TestAccDataSource_basic (8.36s)
=== RUN   TestAccDataSource_updated
--- PASS: TestAccDataSource_updated (13.81s)
=== RUN   TestAccMonitoringJob_basic
--- PASS: TestAccMonitoringJob_basic (6.72s)
=== RUN   TestAccMonitoringJob_updated
--- PASS: TestAccMonitoringJob_updated (8.89s)
=== RUN   TestAccNotifyList_basic
--- PASS: TestAccNotifyList_basic (6.40s)
=== RUN   TestAccNotifyList_updated
--- PASS: TestAccNotifyList_updated (9.92s)
=== RUN   TestAccRecord_basic
--- PASS: TestAccRecord_basic (16.56s)
=== RUN   TestAccRecord_updated
--- PASS: TestAccRecord_updated (20.44s)
=== RUN   TestAccTeam_basic
--- PASS: TestAccTeam_basic (6.78s)
=== RUN   TestAccTeam_updated
--- PASS: TestAccTeam_updated (8.51s)
=== RUN   TestAccUser_basic
--- PASS: TestAccUser_basic (12.74s)
=== RUN   TestAccZone_basic
--- PASS: TestAccZone_basic (7.93s)
=== RUN   TestAccZone_updated
--- PASS: TestAccZone_updated (11.31s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/ns1	170.752s
```

**Comment by pashap** [comment]
@kian would you mind making a new issue for that? thank you

**Pull request merged**